### PR TITLE
migrate to Qt6, Botan 3, make it work on macOS

### PR DESCRIPTION
**Pull request by jlopez**
- **fix: switch to qmake**
- **fix: seek_dir -> seekdir**
- **fix: botan 3 include/lib locations**
- **fix: ensure c++20 std**
- **fix: migrate to botan-3**
- **fix: ensure c++20 std**
- **fix: remove softkeys from mainwindow.ui**
- **fix: case insensitive issue on top Makefile**
- **fix: remove unnecessary botan initialization**
- **fix: add .qmake.stash to .gitignore**
- **fix: pixmap is no longer a pointer**
- **fix: QRegExp to QRegularExpression**
- **fix: switch to UTF-8 encoding**
- **fix: update QDate methods**
- **fix: setItemHidden -> setHidden**
- **fix: add include for QStandardPaths**
- **fix: implicit QChar assignment from int**
- **fix: QNetworkInformation reachability**
- **fix: QRegularExpression new API**
- **fix: use std::sort instead of qSort**
- **fix: QByteArray and QString interactions**
- **fix: QString sprintf deprecation**
- **fix: rename of QPalette::background() -> window()**
- **fix: update QDate methods (tbs)**
- **fix: missing imports**
- **fix: QFontMetrics::width -> horizontalAdvance**
- **fix: QPainter::drawRoundRect -> drawRoundedRect**
- **fix: ensure unix botan libs not used on mac**
- **fix: init QNetworkInformation to avoid crash**
- **chore: add vscode config files**

### CODE REVIEW

**Comment by jlopez** [comment]
Address issue #95

File: COMPILING.md
Comment (hetelek) [inline COMPILING.md:1]: Thanks for doing this! Could you update this COMPILING.md to include updated instructions? Seems it's referencing Qt 4.8.1

Comment (jlopez) [reply COMPILING.md:1]: Just pushed a new commit updating COMPILING.md

Comment (admiralspeedy) [reply COMPILING.md:1]: > Just pushed a new commit updating COMPILING.md

The link for the precompiled Botan library in that readme is still for the old version.

**Pull request merged**